src/frontend/hooks/useSecureStorage.ts

```diff
@@ -0,0 +1,39 @@
+import {setItemAsync, getItemAsync} from 'expo-secure-store';
+import {useState, useEffect, useCallback} from 'react';
+
+type SecureStoreKey = 'RootKey';
+
+export function useSecureStore(key: SecureStoreKey) {
+  const [value, setValue] = useState<string | null>(null);
+  const [status, setStatus] = useState<'idle' | 'loading'>('loading');
+
+  useEffect(() => {
+    getItemAsync(key)
+      .then(storageVal => {
+        setValue(storageVal);
+      })
+      .finally(() => {
+        setStatus('idle');
+      });
+  }, [key, getItemAsync]);
+
+  const setToSecureStore = useCallback(
+    (val: string) => {
+      setStatus('loading');
+      console.log(val);
+      setItemAsync(key, val)
+        .then(() => {
+          setValue(val);
+        })
+        .catch(err => {
+          throw new Error(err);
+        })
```

Comment:
thoughts on introducing an `'error'` status and updating this to set that instead of throwing here? then that error status can be used to show some default error screen or something similar when this hook is used. note that the `finally` would have to be removed/updated if you do change this.

leaving this as is would require having the error boundary set up (which i don't think it is yet) or making sure you wrap any call to `setToSecureStore()` to catch and handle this thrown error

Reply:
I was just using it for trying to debug something, I didn't mean for it to end up in the code. Lets leave out the error for now? Maybe lets talk next week about adding some sort of error boundry. Now would be the time to do it given the state of the app.